PowerShell Scanners/Get-ChocolateyPackages/Get-ChocolateyPackageData.ps1

```diff
@@ -0,0 +1,77 @@
+[cmdletBinding()]
+param(<#reserved for future use#>)
+
+begin {
+    try {
+        Get-Command choco -ErrorAction Stop 
+    } catch {
+        throw "choco not installed, or not found on PATH."
+    }
+}
+
+process {
+
+    function ConvertTo-ChocoObject {
+        <#
+            .SYNOPSIS
+            Converts the output from choco into a pscustomobject
+            
+            .PARAMETER InputObject
+            The pipeline input to process
+            
+            .PARAMETER Audit
+            If you want more detailed information, pass this to get extra audit details
+
+            .EXAMPLE
+            choco list -lo -r | ConvertTo-ChocoObject
+
+            .EXAMPLE
+            choco list -lo -r --audit | ConnvertTo-ChocoObject -Audit
+            #>
+        [CmdletBinding()]
+        Param (
+            [Parameter(ValueFromPipeline)]
+            [string]
+            $InputObject,
+    
+            [Parameter()]
+            [switch]
+            $Audit
+        )
+    
+        Process {
+            if (-not [string]::IsNullOrEmpty($InputObject)) {
```

Comment:
Is there an advantage to doing it this way over `if ($InputObject)`?

Reply:
I _think_ there's a difference between a $null string and an empty string, and doing `if($InputObject)` only satisfies one of them. I'm not 100% sure, and this was taken from our codebase and "modified heavily". So I'm in the "if it ain't broke don't fix it" camp on this one 😂